ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/tls/TlsConfiguration.java

```diff
@@ -19,61 +19,48 @@
 
 import java.nio.file.Path;
 import java.util.Optional;
+import java.util.function.Supplier;
+
+import com.google.common.annotations.VisibleForTesting;
 
 public class TlsConfiguration {
   private final Path keyStorePath;
-  private final String keyStorePassword;
+  private final Supplier<String> keyStorePasswordSupplier;
   private final Path knownClientsFile;
 
   private TlsConfiguration(
-      final Path keyStorePath, final String keyStorePassword, final Path knownClientsFile) {
+      final Path keyStorePath,
+      final Supplier<String> keyStorePasswordSupplier,
+      final Path knownClientsFile) {
     requireNonNull(keyStorePath, "Key Store Path must not be null");
-    requireNonNull(keyStorePassword, "Key Store password must not be null");
+    requireNonNull(keyStorePasswordSupplier, "Key Store password supplier must not be null");
     this.keyStorePath = keyStorePath;
-    this.keyStorePassword = keyStorePassword;
+    this.keyStorePasswordSupplier = keyStorePasswordSupplier;
     this.knownClientsFile = knownClientsFile;
   }
 
+  @VisibleForTesting
```

Comment:
nit: does this help - can the tests just use the fromKeyStoreAndKnownClientConfigurations, and pass in null?

Reply:
👍